### DISCUSSION TITLE
Flora Codex as a temporary alternative to Trefle?

### DISCUSSION THREAD

**Discussion by cbarond** [MDeLuise/plant-it#232] [General]
While researching the shutdown of Trefle, I came across this API called [Flora Codex](https://floracodex.com/). I saw that @MDeLuise is looking at creating a replacement for Trefle (https://github.com/MDeLuise/plant-it/issues/218#issuecomment-2164571002), but this is likely a very large project. In the meantime, would Flora Codex be a possible alternative during development? It seems to be a (mostly) drop-in replacement for Trefle.

I'm curious if anyone has heard about it.
 
Quote from the About page:
"Flora Codex was born when Trefle.io shut down without warning. As a result, we've been scrambling to stand up its replacement -- An even more robust, next generation botanical API. For now we will try to use the Trefle request and response models so that you can use this service as a "drop in" replacement without rewriting your apps."

Notes:
- Free
- Currently in Beta
- Rate limit is 60 requests per minute

**Comment by MDeLuise** [comment]
Hi @cbarond,

Thank you very much for the suggestion!

Flora Codex seems to be a great alternative. I've started testing it, and it appears to be a good replacement, even though it lacks some features such as:

- Plant care information
- Response pagination
- Synonyms

Additionally, it has fewer plants compared to Trefle. 

Despite these limitations, it could serve as an excellent temporary solution. I'll continue testing the service and keep you updated on the progress.

Thanks again for your input!

**Comment by astappiev** [comment]
Interesting finding, it looks like someone was more successful in deploying Trefle from sources than me :)
